### PR TITLE
Add NuGet and .Net Core tasks

### PR DESCRIPTION
**Pull request by barbelity**

### CODE REVIEW

File: src/main/java/org/jfrog/bamboo/configuration/ArtifactoryDotNetCoreConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        super.populateContextForCreate(context);
+        context.put("artifactoryDotNetCoreTask", this);
+        context.put("builderType", this);
+        context.put("builder", this);
+    }
+
+    @Override
+    public void populateContextForEdit(@NotNull Map<String, Object> context, @NotNull TaskDefinition taskDefinition) {
+        super.populateContextForEdit(context, taskDefinition);
+    }
```
Comment (yahavi) [inline src/main/java/org/jfrog/bamboo/configuration/ArtifactoryDotNetCoreConfiguration.java:34]:
```suggestion
```

File: src/main/java/org/jfrog/bamboo/configuration/ArtifactoryNugetConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        super.populateContextForCreate(context);
+        context.put("artifactoryNugetTask", this);
+        context.put("builderType", this);
+        context.put("builder", this);
+    }
+
+    @Override
+    public void populateContextForEdit(@NotNull Map<String, Object> context, @NotNull TaskDefinition taskDefinition) {
+        super.populateContextForEdit(context, taskDefinition);
+    }
```
Comment (yahavi) [inline src/main/java/org/jfrog/bamboo/configuration/ArtifactoryNugetConfiguration.java:34]:
```suggestion
```

File: src/main/java/org/jfrog/bamboo/context/DotNetBuildContext.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    public String getPushPattern() {
+        return env.get(PUSH_PATTERN);
+    }
+
+    public String getPushTarget() {
+        return env.get(PUSH_TARGET);
+    }
+
+    public boolean isRestoreCommand() {
+        return (AbstractDotNetBuildConfiguration.CFG_COMMAND_RESTORE.equals(env.get(COMMAND_CHOICE)));
```
Comment (yahavi) [inline src/main/java/org/jfrog/bamboo/context/DotNetBuildContext.java:60]:
```suggestion
        return AbstractDotNetBuildConfiguration.CFG_COMMAND_RESTORE.equals(env.get(COMMAND_CHOICE));
```

File: src/main/java/org/jfrog/bamboo/task/ArtifactoryDotNetCoreTask.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    private static final String DOTNETCORE_KEY = "system.builder.dotnet.";
+
+    public ArtifactoryDotNetCoreTask(EnvironmentVariableAccessor environmentVariableAccessor, final CapabilityContext capabilityContext) {
+        super(environmentVariableAccessor, capabilityContext);
+    }
+
+    @Override
+    protected void initTask(@NotNull CommonTaskContext context) throws TaskException {
+        taskType = TaskType.DOTNET;
+        super.initTask(context, DOTNETCORE_KEY, EXECUTABLE_NAME, TASK_NAME);
```
Comment (yahavi) [inline src/main/java/org/jfrog/bamboo/task/ArtifactoryDotNetCoreTask.java:24]: Lets pass taskType argument to the initTask method:
```suggestion
        super.initTask(context, DOTNETCORE_KEY, EXECUTABLE_NAME, TASK_NAME, TaskType.DOTNET);
```
Lets do the same in `ArtifactoryDotNetCoreTask`

File: src/main/java/org/jfrog/bamboo/task/ArtifactoryDotNetTaskBase.java [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+        Build build;
+        if (dotNetBuildContext.isRestoreCommand()) {
+            build = executeRestore();
+        } else {
+            try {
+                build = executePush();
+            } catch (Exception e) {
+                String message = "Exception occurred while executing task";
+                logger.addErrorLogEntry(message, e);
+                log.error(message, e);
```
Comment (yahavi) [inline src/main/java/org/jfrog/bamboo/task/ArtifactoryDotNetTaskBase.java:80]: `buildInfoLog.error(...)` does exactly the same
```suggestion
                buildInfoLog.error("Exception occurred while executing task", e);
```

**Review by yahavi — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 6]
```diff
@@ -196,6 +203,12 @@
             <artifactId>atlassian-bamboo-web</artifactId>
             <version>${bamboo.version}</version>
             <scope>provided</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.codehaus.woodstox</groupId>
```
[the remark this answers is not in the document]
Comment (barbelity) [reply pom.xml:208]: Added.

File: src/main/java/org/jfrog/bamboo/configuration/AbstractDotNetBuildConfiguration.java [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+        long serverId = buildContext.getResolutionArtifactoryServerId();
+        if (serverId == -1) {
+            buildContext.resetResolverContextToDefault();
+        }
+    }
+
+    @Override
+    protected String getKey() {
+        return null;
+    }
```
[the remark this answers is not in the document]
Comment (barbelity) [reply src/main/java/org/jfrog/bamboo/configuration/AbstractDotNetBuildConfiguration.java:110]: Removed.

File: src/main/java/org/jfrog/bamboo/configuration/ArtifactoryNugetConfiguration.java (hunk 2 above)
Comment (barbelity) [reply src/main/java/org/jfrog/bamboo/configuration/ArtifactoryNugetConfiguration.java:34]: Removed.

File: src/main/java/org/jfrog/bamboo/task/ArtifactoryDotNetCoreTask.java (hunk 4 above)
Comment (barbelity) [reply src/main/java/org/jfrog/bamboo/task/ArtifactoryDotNetCoreTask.java:24]: Changed.

File: src/main/java/org/jfrog/bamboo/task/ArtifactoryDotNetTaskBase.java (hunk 5 above)
Comment (barbelity) [reply src/main/java/org/jfrog/bamboo/task/ArtifactoryDotNetTaskBase.java:80]: Fixed.

**Pull request merged**